pkg/v1/tkg/client/scale_test.go

```diff
@@ -123,86 +127,181 @@ var _ = Describe("Scale API", func() {
+					It("should not update any mds", func() {
+						scaleClusterOptions.WorkerCount = 0
+
+						err = tkgClient.DoScaleCluster(regionalClusterClient, &scaleClusterOptions)
+						Expect(err).ToNot(HaveOccurred())
+
+						Expect(regionalClusterClient.GetMDObjectForClusterCallCount()).To(Equal(0))
+						Expect(regionalClusterClient.UpdateReplicasCallCount()).To(Equal(0))
+					})
 				})
 			})
-			When("a user calls ScaleCluster with a worker count less than 1", func() {
-				It("should not update any mds", func() {
-					regionalClusterClient.IsPacificRegionalClusterReturns(false, nil)
-					regionalClusterClient.GetKCPObjectForClusterReturns(nil, nil)
+			Context("and scale is operating on a specific node pool", func() {
+				When("a user scales an existing node pool", func() {
+					It("should update the replicas of that node pool", func() {
+						regionalClusterClient.GetMDObjectForClusterReturnsOnCall(0, []v1alpha3.MachineDeployment{md1, md2, md3}, nil)
+						regionalClusterClient.GetMDObjectForClusterReturnsOnCall(1, []v1alpha3.MachineDeployment{md1, md2, md3}, nil)
+
+						scaleClusterOptions.NodePoolName = md1Name
+						scaleClusterOptions.WorkerCount = 3
+
+						err = tkgClient.DoScaleCluster(regionalClusterClient, &scaleClusterOptions)
+						Expect(err).ToNot(HaveOccurred())
+
+						Expect(regionalClusterClient.GetMDObjectForClusterCallCount()).To(Equal(2))
+						Expect(regionalClusterClient.UpdateResourceCallCount()).To(Equal(1))
+						mdInterface, _, _, _ := regionalClusterClient.UpdateResourceArgsForCall(0)
+						md := mdInterface.(*v1alpha3.MachineDeployment)
+						Expect(*md.Spec.Replicas).To(Equal(int32(3)))
+					})
+				})
+				When("an error occurs retrieving machine deployments", func() {
+					It("should throw an error", func() {
+						mdErrString := "failed retrieving mds"
+						regionalClusterClient.GetMDObjectForClusterReturnsOnCall(0, nil, errors.New(mdErrString))
+
+						scaleClusterOptions.NodePoolName = md1Name
+						scaleClusterOptions.WorkerCount = 3
+
+						err = tkgClient.DoScaleCluster(regionalClusterClient, &scaleClusterOptions)
+						Expect(err).Should(MatchError(fmt.Sprintf("Failed to get node pools for cluster %s: unable to get worker machine deployments: %s", scaleClusterOptions.ClusterName, mdErrString)))
+					})
+				})
+				When("the named node pool can't be found", func() {
+					It("should throw an error", func() {
+						regionalClusterClient.GetMDObjectForClusterReturnsOnCall(0, nil, nil)
+
+						scaleClusterOptions.NodePoolName = md1Name
+						scaleClusterOptions.WorkerCount = 3
+
+						err = tkgClient.DoScaleCluster(regionalClusterClient, &scaleClusterOptions)
+						Expect(err).Should(MatchError(fmt.Sprintf("Could not find node pool with name %s", scaleClusterOptions.NodePoolName)))
+					})
+				})
+				When("the node pool fails to be updated", func() {
+					It("should update the replicas of that node pool", func() {
```

Comment:
nit: should it be `should not update the replicas of that node pool` or `should return error`?